futures-util/src/async_await/random.rs

```diff
@@ -18,18 +18,15 @@ pub fn shuffle<T>(slice: &mut [T]) {
 
 /// Return a value from `0..n`.
 fn gen_index(n: usize) -> usize {
-    (random() % n as u64) as usize
+    random() % n
 }
 
 /// Pseudorandom number generator based on [xorshift*].
 ///
 /// [xorshift*]: https://en.wikipedia.org/wiki/Xorshift#xorshift*
```

Comment:
Since different pointer widths use different algorithms it would be good to move this comment to them. It looks like it's currently `xorshift64*`, `xorshift32` and non-standard `xorshift16` (is there a reference for the constants chosen for this? the xorshift paper only appears to have tables for 32 and 64).

Reply:
Yeah, this should definitely be documented and a conscious decision made for each. I picked something random just to get the concept up to show, and to discuss whether plain xorshift32 is "good enough" (it even feels like `xorshift64*` is overkill but I left it as-is considering it's the existing implementation and unlikely to matter on 64bit targets anyway). The most relevant discussion of 16bit variants I've come across is [here](http://b2d-f9r.blogspot.com/2010/08/16-bit-xorshift-rng-now-with-more.html), my target platform in mind here is msp430 (are there any other 16bit rust targets?).